include/swift/AST/DiagnosticsSema.def

```diff
@@ -1514,6 +1514,9 @@ ERROR(access_control_extension_open,none,
 ERROR(access_control_open_bad_decl,none,
       "only classes and overridable class members can be declared 'open';"
       " use 'public'", ())
+WARNING(access_control_non_objc_open_member,none,
+      "non-'@objc' %select{properties|methods}0 in extensions cannot be overridden; use 'public' instead",
+      (bool))
```

Comment:
To address @xwu comments about subscript being mentioned as property, we could use `DescriptiveDeclKind` here. 
```suggestion
      "non-'@objc' %0 in extensions cannot be overridden; use 'public' instead",
      (DescriptiveDeclKind))
```
And pass it as `VD->getDescriptiveKind()`. I think that would be a simpler way to address it :)

Reply:
Thanks for the guide! Fixed it :)